book/actix-plugin.md

```diff
@@ -156,6 +156,55 @@ curl http://localhost:8080/api/spec
 
 Similarly, if we were to use other extractors like `web::Query<T>`, `web::Form<T>` or `web::Path`, the plugin will emit the corresponding specification as expected.
 
+#### Manually defining additional response codes
+
+There is a macro `api_v2_errors` which helps to manually add responses other than 200. Here is an example.
+We define an enum that represents the error. To use it in actix-web it needs to implement `Display` and `ResponseError`.
+There is no need to impl Display by hand for such simple enums - we can use `enum-display-derive` lib.
+
+```rust
+use actix_web::{
+    error::{ErrorInternalServerError, ErrorUnauthorized, ResponseError},
+    HttpResponse,
+};
+use enum_display_derive::Display;
+use paperclip::actix::api_v2_errors;
+use std::fmt::Display;
+
+#[api_v2_errors(
+    "400 Bad Request",
+    "401 Unauthorized: Can't read session from header",
+    "500 Internal Server Error",
```

Comment:
I'm wary of using strings here. Firstly, can we make this a bit more readable?

```rust
#[api_v2_errors(
    code = 400, description = "Bad Request",
    code = 500, description = "Internal Server Error",
)]
```

I also don't want to enforce the description here. Users can avoid descriptions, in which case we'd fill the description of the status code automatically (from say, `http` crate).

```rust
#[api_v2_errors(
    code = 400, // description will default to "Bad Request"
    code = 500, description = "Booya",
)]
```

Users could also choose to not have any descriptions, in which case, they could go for a list of status codes.

```rust
#[api_v2_errors(400, 500)]
```

This is what I have in mind. If you have a suggestion, then I'm all ears! :smile: